apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/FissClaimTransformerV2.java

```diff
@@ -282,6 +320,132 @@ private static List<Claim.InsuranceComponent> getInsurance(RdaFissClaim claimGro
             })
         .filter(Objects::nonNull)
         .sorted(Comparator.comparing(Claim.InsuranceComponent::getSequence))
-        .collect(Collectors.toList());
+        .toList();
+  }
+
+  /**
+   * Maps {@link RdaFissRevenueLine} data to appropriate FHIR components.
+   *
+   * @param claimGroup The claim data to map.
+   * @return The created FHIR component with given claim data.
+   */
+  private static List<Claim.ItemComponent> getClaimItems(RdaFissClaim claimGroup) {
+    return claimGroup.getRevenueLines().stream()
+        .map(
+            revenueLine -> {
+              Claim.ItemComponent itemComponent = new Claim.ItemComponent();
+
+              itemComponent.setSequence(revenueLine.getRdaPosition());
+
+              CodeableConcept revenue;
+
+              if (Strings.isNotBlank(revenueLine.getNonBillRevCode())
+                  || Strings.isNotBlank(revenueLine.getRevCd())) {
+                revenue = new CodeableConcept();
+
+                if (Strings.isNotBlank(revenueLine.getNonBillRevCode())) {
+                  revenue.setCoding(
+                      List.of(
+                          new Coding(
+                              BBCodingSystems.FISS.REV_CD, revenueLine.getNonBillRevCode(), null)));
+                }
+
+                if (Strings.isNotBlank(revenueLine.getRevCd())) {
+                  revenue.addExtension(
+                      BBCodingSystems.FISS.NON_BILL_REV_CODE,
+                      new CodeableConcept(
+                          new Coding(
+                              BBCodingSystems.FISS.NON_BILL_REV_CODE,
+                              revenueLine.getRevCd(),
+                              null)));
+                }
+              } else {
+                revenue = null;
+              }
+
+              itemComponent.setRevenue(revenue);
```

Comment:
We set `revenue` to null if there is no code but below we set `itemComponent` to an blank `CodeableConcept` if there is no `hcpcCd` or `apcHcpcCd`.  Is that inconsistency intentional?

Reply:
The code handling `hcpcCd` and `apcHcpcCd` is checking if at least one of them is not blank, only creating a `CodeableConcept` if it actually has a value added to it, which is the same as for `revenue`, however I do see how it looked different as I was handling the setter at different scopes between the two examples.  Functionally the same, but syntactically different, I'll make that more consistent for readability.